lib/index.js

```diff
@@ -1,78 +1,84 @@
 const Iterable = (obj) => {
   const isobject =
-    (obj) =>
-      ( typeof obj === 'object' && ({}).toString.call(obj) === '[object Object]' )
+  (obj) => (typeof obj === 'object' && ({}).toString.call(obj) === '[object Object]')
 
   if (!isobject(obj)) return
 
   var cache = Object.keys(obj)
   var l = cache.length
 
-  function each(cb) {
+  function each (cb) {
     if (!l) return
     cache.forEach((k) => {
       cb(obj[k], k, obj)
     })
     return obj
   }
 
-  function map(cb) {
+  function map (cb) {
     if (!l) return
-    return Iterable( cache.reduce((acc, k) => {
+    return Iterable(cache.reduce((acc, k) => {
       acc[k] = cb(obj[k], k, o) || obj[k]
-      return acc;
-    }, {}) )
+      return acc
+    }, {}))
   }
 
-  function filter(cb) {
+  function filter (cb) {
     if (!l) return
-    return Iterable( cache.reduce((acc, k) => {
+    return Iterable(cache.reduce((acc, k) => {
       if (cb(obj[k], k, o))
-      acc[k] = obj[k]
-      return acc;
-    }, {}) )
+        acc[k] = obj[k]
+      return acc
+    }, {}))
   }
 
-  function updateCache(key) {
+  function updateCache (key) {
     cache.push(key)
     l = cache.length
   }
 
-  function set(k, v) {
+  function set (k, v) {
     updateCache(k)
     obj[k] = v
     return obj
   }
 
-  function kill(prop) {
+  function kill (prop) {
     if (!l) return
     if (!(prop in obj)) return
     delete obj[prop]
     return obj
   }
 
-  function force() {
+  function force () {
     cache = Object.keys(obj)
     l = cache.length
     return obj
   }
 
-  function forceKill(prop) {
+  function forceKill (prop) {
     kill(prop)
     return force()
   }
 
+  function returnIterable () {
+    return (function * (obj) {
+      for (let key of Object.keys(obj)) yield ({key: key, value: obj[key]})
```

Comment:
true that, but generator felt like obvious way to go. maybe I should reimplement it with a simple iterator solution?